index.bs

```diff
@@ -1470,6 +1471,24 @@ To <dfn>parse a filter pair</dfn> given a [=map=] |map|:
 1. If |negative| is null, return null.
 1. Return the [=tuple=] (|positive|, |negative|).
 
+<h3 id="getting-preferred-platform">Getting preferred platform</h3>
+
+To <dfn>get preferred platform from a header list</dfn> given a
+[=header list=] |headers|:
+
+1. If |headers| does not [=header list/contain=] "`Attribution-Reporting-Info`", return null.
+1. Let |map| be the result of
+    [=header list/get a structured field value|getting=] "`Attribution-Reporting-Info`" from |headers|
+    with a type of "`dictionary`".
+1. If |map| is not an [=ordered map=], return an error.
+1. If |map|["`preferred_platform`"] does not [=map/exist=], return null.
+1. Let |value| be |map|["`preferred_platform`"][0].
+1. If |value| is not a [=string=], return an error.
```

Comment:
This step is redundant with the `registrar` check below, especially since we have an issue to ensure this is a token.